dnsapi/dns_freedns.sh

```diff
@@ -245,6 +214,73 @@ _freedns_login() {
   return 0
 }
 
+# usage: _freedns_find_domain "<FREEDNS_COOKIE>" "www.domain.com"
+# print DNSdomainid on success
+# returns 0 on success
+_freedns_find_domain() {
+  FREEDNS_COOKIE="$1"
+  top_domain="$2"
+
+  # Sometimes FreeDNS does not return the subdomain page but rather
+  # returns a page regarding becoming a premium member.  This usually
+  # happens after a period of inactivity.  Immediately trying again
+  # returns the correct subdomain page.  So, we will try twice to
+  # load the page and obtain our domain ID
+  attempts=2
+  while [ "$attempts" -gt "0" ]; do
+    attempts="$(_math "$attempts" - 1)"
+
+    htmlpage="$(_freedns_retrieve_subdomain_page "$FREEDNS_COOKIE")"
+    if [ "$?" != "0" ]; then
+      if [ "$using_cached_cookies" = "true" ]; then
+        _err "Has your FreeDNS username and password changed?  If so..."
+        _err "Please export as FREEDNS_User / FREEDNS_Password and try again."
+      fi
+      return 1
+    fi
+
+    subdomain_csv="$(echo "$htmlpage" | tr -d "\n\r" | _egrep_o '<form .*</form>' | sed 's/<tr>/@<tr>/g' | tr '@' '\n' | grep edit.php | grep "$top_domain")"
```

Comment:
if you trim `\r\n` before  piping to `_egrep_o()` function,  it will not work on solaris.

On solaris,  `_egrep_o` is implemented by `sed`, but `sed` on solaris can not work without `\n`.

please consider replacement